src/handlers/generate-payout-permit.ts

```diff
@@ -19,10 +19,10 @@ export async function generatePayoutPermit(context: Context, permitRequests: Per
     let permit: Permit;
     switch (type) {
       case "ERC20":
-        permit = await generateErc20PermitSignature(context, username, amount);
+        permit = await generateErc20PermitSignature(username, amount, context);
```

Comment:
Only reason I pushed it last is because of the function overload, would be impossible to make the function signature work. Thing is JavaScript does not support function overload by itself, to handle this optional arguments of the signature need to be passed last.
If we really want to overcome this I can directly overload by making a `Payload` object, I will change it then.

Reply:
I agree that optional parameters should be placed last in typescript

Reply:
All right, thank you for the explanation, lets merge this as is then.